src/ComponentBase.ts

```diff
@@ -67,22 +66,27 @@ export abstract class ComponentBase<P = {}, S = {}> extends React.Component<P, S
         this.state = {
             _resubGetInstance: () => instance,
             _resubDirty: false,
-        } as unknown as ComponentBaseState<S>;
+        } as InternalState as unknown as S;
     }
 
     // Subclasses may redeclare, but must call ComponentBase.getDerivedStateFromProps
-    static getDerivedStateFromProps: React.GetDerivedStateFromProps<any, ComponentBaseState<any>> =
-    (nextProps, prevState: ComponentBaseState<unknown>) => {
-        if(prevState) {
-            let instance = prevState._resubGetInstance();
-            if(instance) {
-                if(!instance._isMounted) {
-                    return instance._buildInitialState();
-                }
-                return instance._handleUpdate(nextProps);
-            }
+    static getDerivedStateFromProps: React.GetDerivedStateFromProps<unknown, unknown> =
+    (nextProps, prevState: unknown) => {
+        const internalState = prevState as InternalState;
+        if (!internalState._resubGetInstance) {
+            throw new Error("Resub internal state missing - ensure you aren't setting state directly in component construtor")
         }
-        throw new Error('couldn\'t get instance of Component');
+        let newState: unknown & Partial<InternalState>;
+        let instance = internalState._resubGetInstance();
+        if(!instance._isMounted) {
```

Comment:
Somehow our lint rules are letting this through.  Should investigate why at some point.